tests/api_tests/users/test_views.py

```diff
@@ -203,6 +204,275 @@ def test_get_projects_logged_in_as_different_user(self):
+            'github': self.newGithub,
+            'personal_website': self.newPersonal_website,
+            'twitter': self.newTwitter,
+            'linkedIn': self.newLinkedIn,
+            'impactStory': self.newImpactStory,
+            'orcid': self.newOrcid,
+            'researcherId': self.newResearcherId,
+        }
+        self.blank_user_data =  {
+            'id': self.user_one._id,
+            'fullname': self.new_fullname,
+            'given_name': '',
+            'family_name': '',
+            'suffix': '',
+            'github': '',
+            'personal_website': '',
+            'twitter': '',
+            'linkedIn':  '',
+            'impactStory': '',
+            'orcid': '',
+            'researcherId': '',
+        }
+
+    def test_patch_user_logged_out(self):
+        res = self.app.patch_json(self.user_one_url, {
+            'fullname': self.new_fullname,
+        }, expect_errors=True)
+        # This is 403 instead of 401 because basic authentication is only for unit tests and, in order to keep from
+        # presenting a basic authentication dialog box in the front end. We may change this as we understand CAS
+        # a little better
+        assert_equal(res.status_code, 403)
+
+    def test_patch_user_read_only_field(self):
+        # Logged in user updates their user information via patch
+        res = self.app.patch_json(self.user_one_url, {
+            'employment_institutions': self.new_employment_institutions,
+            'educational_institutions': self.new_educational_institutions,
+            'fullname': self.new_fullname,
+
+        }, auth=self.auth_one)
+        assert_equal(res.status_code, 200)
+        assert_equal(res.json['data']['employment_institutions'], self.user_one.jobs)
+        assert_equal(res.json['data']['educational_institutions'], self.user_one.schools)
+        assert_equal(res.json['data']['fullname'], self.new_fullname)
+
+    def test_put_user_logged_in(self):
+        # Logged in user updates their user information via patch
+        res = self.app.put_json(self.user_one_url, self.new_user_data, auth=self.auth_one)
+        assert_equal(res.status_code, 200)
+        assert_equal(res.json['data']['fullname'], self.new_fullname)
+        assert_equal(res.json['data']['given_name'], self.new_given_name)
+        assert_equal(res.json['data']['family_name'], self.new_family_name)
+        assert_equal(res.json['data']['suffix'], self.new_suffix)
+        assert_equal(res.json['data']['github'], self.newGithub)
+        assert_equal(res.json['data']['personal_website'], self.newPersonal_website)
+        assert_equal(res.json['data']['twitter'], self.newTwitter)
+        assert_equal(res.json['data']['linkedIn'], self.newLinkedIn)
+        assert_equal(res.json['data']['impactStory'], self.newImpactStory)
+        assert_equal(res.json['data']['orcid'], self.newOrcid)
+        assert_equal(res.json['data']['researcherId'], self.newResearcherId)
```

Comment:
Can you add assertions to ensure that the user was correctly updated in the database?